src/components/split_button/split_button_control.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, {
+  FunctionComponent,
+  ButtonHTMLAttributes,
+  ReactNode,
+} from 'react';
+// import classNames from 'classnames';
+
+import { CommonProps } from '../common';
+
+// import { htmlIdGenerator } from '../../services/accessibility';
+// import { OuiFormControlLayoutProps } from '../form/form_control_layout';
+// import { OuiFlexGroup, OuiFlexItem } from '../flex';
+import {
+  ButtonColor,
+  ButtonSize,
+  OuiButton,
+  OuiButtonIcon,
+  OuiButtonIconColor,
+} from '../button';
+
+export type OuiSplitButtonOption = string | ReactNode;
+
+export interface OuiSplitButtonControlProps
+  extends CommonProps,
+    ButtonHTMLAttributes<HTMLButtonElement> {
```

Comment:
if I have 
`interface OuiSplitButtonControlProps { propsForPrimaryButton: ??,  propsForDropdownButton: ??}`
What type would you use for these? (`??`)

Reply:
This would actually be a case where we use `ExclusiveUnion`, when ensures you only use props from one type or the other type - you can't mix props from both types. In this scenario, you can use `PropsForAnchor` and `PropsForButton`, similar to what `OuiButton` does. Here are the docs:

https://github.com/opensearch-project/oui/blob/6ef0c944d690719f9ff9093b8374a684dab01e79/src/components/common.ts#L184-L205